### DISCUSSION TITLE
Ironbar launcher and special workspaces.

### DISCUSSION THREAD

**Discussion by leoVici9** [JakeStanger/ironbar#815] [General]
Have started using scratchpads based on special workspaces a lot lately . For  example fir a TUI file manager and clipboard manager. However the behaviour of  the launcher widget seems not inline with this approach wanted to start this discussion as feedback and to see how this can be improved 
1 ) Basically the launcher displays applications in a scratchpad  and clicking on the application icon will open the scratchpad.  When there are multiple applications in a scratchpad this becomes very confusing.   Would it be possible to hide applications that are in s scratch pad in the launcher and only show them when the scratch pad is visible ? 
2) to get hyprland window rules and TUI applications working the use of app-id on starting the terminal is very convenient. However launcher only displays proper icons for known classes. Starting a terminal with a app-id like dropterm launcher would  display unknown icon. Would be nice if an icon mapping could somehow be specified 
3) if implementing scratch's by using a third-party program (pyprland) clicking on the icon in the launcher will display the application window at wrong locations (pyprland specifies it's own window rules which are not known to hyprland, so restoring a scratch pad through the launcher will by pass pyprland) 

Any suggestions ? Comments ?

**Comment by JakeStanger** [comment]
Thought I'd replied to this but must have typed out half of it and got distracted...

1. Currently not possible, but I would like to add a feature to the launcher to only show programs open on the current display/workspace. This is somewhat pending support for the new `ext-workspace` protocol.
2. Yes, this is a feature I've had in the back of my mind for a very long time. 
3. Unsure on this one.

For 1 & 2, feel free to open feature request issues.
